lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -26,17 +26,54 @@ def perform
       end
 
       def entity_table_check_data(model)
+        adapter_name = ActiveRecord::Base.connection.adapter_name.downcase
+
+        if adapter_name != 'postgresql' && Rails.env.production?
+          Rails.logger.info('DfE::Analytics: Entity checksum: Only Postgres databases supported on PRODUCTION')
+          return
+        end
+
+        row_count, checksum, checksum_calculated_at = fetch_checksum_data(model, adapter_name)
+        {
+          row_count: row_count,
+          checksum: checksum,
+          checksum_calculated_at: checksum_calculated_at
+        }
+      end
+
+      def fetch_checksum_data(model, adapter_name)
+        if adapter_name == 'postgresql'
+          fetch_postgresql_checksum_data(model)
+        else
+          fetch_generic_checksum_data(model)
+        end
+      end
+
+      def fetch_postgresql_checksum_data(model)
+        checksum_sql_query = <<-SQL
+          SELECT COUNT(*) as row_count,
+            MD5(STRING_AGG(CHECKSUM_TABLE.ID, '')) as checksum,
+            CURRENT_TIMESTAMP(6) as checksum_calculated_at
+          FROM (
+            SELECT #{model.table_name}.id::TEXT as ID
+            FROM #{model.table_name}
+            WHERE #{model.table_name}.updated_at < CURRENT_TIMESTAMP(6)
+            ORDER BY #{model.table_name}.updated_at ASC
+          ) CHECKSUM_TABLE
+        SQL
+
+        result = ActiveRecord::Base.connection.execute(checksum_sql_query).first
+        [result['row_count'].to_i, result['checksum'], result['checksum_calculated_at']]
+      end
+
+      def fetch_generic_checksum_data(model)
```

Comment:
current_timestamp could be passed into this method rather than calculating. Please see comment above.

Reply:
Now passing in the current_timestamp to both fetches